documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,263 @@
+
+This page should use the
+[`validate` method](https://v2.nuxt.com/docs/components-glossary/validate/) to
+make sure that the `collection`, `searchType` and `term` `params` are valid.
+
+```
+`validate({ params, $pinia }): {
+const { collection, searchType, term } = params
+```
+
+This page should also update the state (`searchType`, `searchTerm` and
+`searchBy` filter) in the `search` store and handle fetching using
+`mediaStore`'s `fetchMedia` method in the `useFetch` hook. Since it is not
+possible to change the path or query parameters from this page client-side,
+fetching can be much simpler than on the current search page (that has to watch
+for changes in the route and fetch if necessary).
+
+This page should have the common header (title, button - link to the source
+external URL or the creator URL, and the summary of item counts in the
+collection), and a nested component: the image grid or the audio collection.
+
+### New and updated components
+
+#### Extract the `VAudioCollection` component
+
+Currently, it is not possible to reuse the audio collection from the audio
+search result page because it is a part of the `audio.vue` page. We should
+extract the part that shows the loading skeleton, the column of `VAudioTrack`
+rows and the Load more section. This component will be reused in the audio
+search page and on the Additional search views. One thing that I'm not sure
+about here is whether the analytics events should stay the same and be sent from
+all pages - or if we should somehow differentiate the events coming from
+different search views.
+
+#### Add a `VCollectionHeader` component
+
+Figma links:
+[creator desktop](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56323&mode=design&t=pN0PPAzlKQEKT9tJ-4),
+[creator mobile](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56362&mode=design&t=suLIyJHNmZrM0mPH-4),
+[source desktop](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56381&mode=design&t=suLIyJHNmZrM0mPH-4),
+[source mobile](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56421&mode=design&t=suLIyJHNmZrM0mPH-4),
+[tag desktop](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1216-58402&mode=design&t=suLIyJHNmZrM0mPH-4),
+[tag mobile](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56457&mode=design&t=suLIyJHNmZrM0mPH-4)
+
+The header should have an icon (tag, creator or source), the name of the
+tag/creator/source. For source and the creator, there should be an external link
+button.
+
+The header should also display the number of results, "251 audio files with the
+selected tag", "604 images provided by this source", "37 images by this creator
+in Hirshhorn Museum and Sculpture Garden".
```

Comment:
Is it at all relevant for us to consider sources that have a single creator? Boston Public Library, as a source, for example, only has Boston Public Library as the "creator" of the work. It's weird to think that there would be two separate pages for them, but then I'm not sure how we would communicate this to the user if we automatically redirected the BPL creator search to the BPL source search :thinking:

Reply:
Added a note to the plan. By the way, we don't have a Boston Public Library as a source filter on the frontend. What is the link to their API or collection?

Reply:
Oops, I've just realised that BPL is just a creator we've picked up through Flickr, not an actual source. Maybe an easy source to add then!